doc/man/man5/strx25.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.TH STRX25 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ Packages"
.SH NAME
.B @PACKAGE@
\- @PACKAGE_TITLE@ Package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.B @PACKAGE_TITLE@
.RB ( @PACKAGE@-@VERSION@ )
package for
.BR "Linux Fast-STREAMS" .
.[
lfs
.]
.PP
OpenSS7
.I STREAMS
add-on packages are separate add-on packages to the
.B "Linux Fast-STREAMS"
.[
lfs
.]
release that provide additional functionality and protocol support not
traditionally found on UNIX base systems, or which have traditionally fallen
into the domain of value-add system software, or, in this case, where the
functionality was optional or differently handled in a number of
implementations.  The
.B @PACKAGE_TITLE@
package is one of several optional add-on packages that provide enhanced
.I STREAMS
support beyond that provided by the base
.I STREAMS
packages.  This package provides support for network protocols, not provided
by the base
.I STREAMS
package.
.\"
.\"
.SH DESCRIPTION
.PP
The
.B @PACKAGE_TITLE@
package provides
.I STREAMS
modules and drivers, libraries, programs, initialization scripts, and daemons
that implement the
.IR "ITU-T Recommendation X.25" ,
.[
x25
.]
.IR "ISO/IEC 8208" ,
.[
iso8208
.]
and
.IR "ISO/IEC 8881" ,
.[
iso8881
.]
data links, networks, interfaces and applications.
.PP
The package currently provides the following
.I STREAMS
modules and drivers:
.PP
Drivers:
.TP
.BR cd-llc (4)
.PD 0
.TP
.BR dl-llc (4)
.TP
.BR ifr (4)
.TP
.BR ixe (4)
.TP
.BR lapb (4)
.TP
.BR llc2 (4)
.TP
.BR mfe (4)
.TP
.BR mlp (4)
.TP
.BR npi (4)
.TP
.BR wan (4)
.TP
.BR x25-lapb (4)
.TP
.BR x25-llc2 (4)
.TP
.BR x25-llc (4)
.TP
.BR x25-plp (4)
.TP
.BR xx25 (4)
.PD
.PP
Modules:
.TP
.BR cdpmod (4)
.PD 0
.TP
.BR dcc (4)
.TP
.BR dlaix (4)
.TP
.BR hdlc (4)
.TP
.BR lapb (4)
.TP
.BR npi (4)
.TP
.BR s_cdi (4)
.TP
.BR s_lli (4)
.TP
.BR s_llic (4)
.TP
.BR s_nli (4)
.TP
.BR s_nlic (4)
.TP
.BR s_wan (4)
.TP
.BR s_wanc (4)
.TP
.BR xos (4)
.TP
.BR xot (4)
.TP
.BR xx25 (4)
.PD
.PP
The package builds and installs the following shared object libraries:
.TP
.B libsx25 (3)
.PP
The
.B @PACKAGE_TITLE@
package includes kernel modules,
.I SVR 4.2
.[
svr42
.]
.I STREAMS
drivers, modules, libraries, utilities, test programs, daemons and development
environment for the development and execution of
.B @PACKAGE_TITLE@
protocol stack components and applications for the
.I Linux Fast-STREAMS
.[
lfs
.]
environment.
.PP
This distribute is only currently applicable to
.BR "Linux 2.4" " and " 2.6
kernels and was targetted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
archictures, but should build and install for other architectures supported by
the Linux kernel as well.
.PP
This package is currently incomplete.  It is being released as a reference
point for the community.  If you are interested in completion of this package,
contact
.RB < info@openss7.com >.
.\"
.\"
.SH PACKAGE CONTENTS
.PP
The
.B @PACKAGE_TITLE@
.RB ( @PACKAGE@-@VERSION@ )
package contains the following manual pages:
.\"
.\"
.SH MODULES
.PP
The
.B @PACKAGE@
package provides the following
.I STREAMS
modules:
.\"
.\"
.SH DRIVERS
.PP
The
.B @PACKAGE@
package provides the following
.I STREAMS
drivers:
.\"
.\"
.SH DEVICES
.PP
The
.B @PACKAGE@
package provides the following
.I STREAMS
devices:
.\"
.\"
.SH PREREQUISITES
.PP
The
.B @PACKAGE@
package requires the following packages to be installed:
.TP \w'\fBstrcompat\fP(5)\(em'u
.BR streams (5)
The
.B Linux Fast-STREAMS
base package,
.BR strss7-@LFS_VERSION@ .
.TP
.BR strcompat (5)
The
.B Linux Fast-STREAMS
compatiblity modules package.
.TP
.BR strxns (5)
The
.B Linux Fast-STREAMS
X/Open Networking Services package.
.TP
.BR strxnet (5)
The
.B Linux Fast-STREAMS
XTI/TLI Library package.
.TP
.BR strnsl (5)
The
.B Linux Fast-STREAMS
Network Selection Library package.
This package is optional.
.TP
.BR strsock (5)
The
.B Linux Fast-STREAMS
STREAMS Sockets Library package.
This package is optional.
.TP
.BR strchan (5)
The
.B Linux Fast-STREAMS
Channel Drivers package.
This package is optional.
.\"
.\"
.SH NOTICES
.PP
The
.B @PACKAGE_TITLE@
package has known and unknonw defects.  This is a
.I pre-alpha
release.  Some defects might be harmful.  No validation testing whatsoever has
been performed by the
.I OpenSS7 Project
on this software.  The software might not even configure or compile.  The
.I OpenSS7 Project
recommends that you do not use this software.  Use at your own risk.  Remember
that there is
.BR "NO WARRANTY" .
(See section
.RI \(lq "NO WARRANTY" \(rg
under the
.RI \(lq "GNU Affero General Public License" \(rq,
in the manual.)
.PP
This software is
.I pre-alpha
software.  As such, it will crash your kernel.  Installation of the software
will irreparably mangle your header files or Linux distribute in such a way as
to make it unusable.  Crashes will lock your system and rebooting the system
will not repair the problem.  You will lose all the data on your system.
Because this software will crash your kenrel, the resulting unstable system
can destroy computer hardware or peripherals making them unusable.  You will
void the warranty on any system on which you run this software.
.BR "YOU HAVE BEEN WARNED" .
.\"
.\"
.SH BUGS
.PP
The
.B @PACKAGE@
package is compatible with a wide array of production
.B Linux
distributions, kernels, and target architectures, but is considered
.I pre-alpha
grade by
.IR "The OpenSS7 Project" .
At the time of rlease, the
.B @PACKAGE@
package had no known bugs.  Problems or development request can be reported to
.RB < bugs@openss7.org >.
Please, however, use the included
.BR send-pr (1)
script for generating actionable problem reports or development requests.
Please see the file
.RI \(lq PROBLEMS \(rq
in the release for problem reporting guidelines.
.\"
.\"
.SH SEE ALSO
.PP
.BR @PACKAGE@_mknod (8),
.IR STREAMS (9).
.PP
Note that, although the modules and drivers provided in the
.B @PACKAGE@
package support the operation of OSI CONS over X.25 as specified in
.IR "ISO/IEC 8878 (X.223)" ,
.[
x223
.]
and the Subnetwork Address Resolution Entity (SNARE) as specified in
.IR "ISO/IEC 10030" ,
.[
iso10030
.]
these additional modules necessary to support CONS are provided in the
.BR striso (5)
package.
.\"
.\"
.SH COMPATIBILITY
.PP
The
.B @PACKAGE@
package is compatible with
.I SVR 4.2
.[
svr42
.]
and X.25 implementations on systems based on
.IR "SVR 4.2" ,
such as
.IR "Solaris" \(tm " Solstice X.25",
.[
solx25dg
.]
.[
solx25ug
.]
.[
solx25pg
.]
.IR "SGI IRIX" \(rg " SX.25" ,
.[
sx25
.]
.IR "PT X.25" ,
.[
ptx25
.]
.IR "AIX" \(rg " AIXlink/X.25" ,
.[
aixx25
.]
with the following portability considerations:
.IP \(em \w'\(em\(em'u
The
.B @PACKAGE_TITLE@
package provides compatibility with all implementations of the X.25 Protocol
suite using the
Network Layer Interface (NLI),
.[
sx25
.]
.BR nli (7),
Data Link Provider Interface (DLPI),
.[
dlpi revision 2.0.0
.]
.BR dlpi (7),
Network Provider Interface (NPI),
.[
npi revision 2.0.0
.]
.BR npi (7),
X/Open Transport Interface X.25 (XX25),
.[
xx25
.]
.BR xti_x25 (3),
specifications.
.IP \(em
These drivers are designed to operate with the XTI/TLI library implementation
provided by the
.BR strxnet (5)
package and is thus compatible with most XTI library implementations as well.
.\"
.\"
.SH CONFORMANCE
.PP
.IR DLPI ,
.[
dlpi revision 2.0.0
.]
.IR NLI ,
.[
sx25
.]
.IR NPI ,
.[
npi revision 2.0.0
.]
.IR XTI ,
.[
xti revision 1.0
.]
.IR "XNS 5.2" ,
.[
xns 5.2
.]
.IR XX25 ,
.[
xx25
.]
.IR UNIX95 ,
.IR UNIX98 .
.[
susv2
.]
Conformance is tested using the
.BR test-x25 (8)
test case executable and the
.I @PACKAGE_TITLE@
.B autotest
test suite.
.\"
.\"
.SH HISTORY
.PP
The contents of the
.B @PACKAGE@
package were formerly part of the
.BR strss7 (5)
package and then the
.BR striso (5)
package.  As of
.BR @PACKAGE@-@VERSION@ ,
the X.25 components were moved out of the
.BR striso (5)
package into this
.BR @PACKAGE@ (5)
package.  This package now contains all of the X.25 protocol components
released by
.IR "The OpenSS7 Project" ,
along with associated documentation.
.PP
In the initial release of the
.BR strx25 (5)
package, only X.25 modules and drivers that have been fully revalidated on
.BR Linux Fast-STREAMS
were initially released.  The remaining components present on the CVS archive
and formerly part of the commercial
.BR striso (5)
releases, will be publicly released in released subsequent to
.B @PACKAGE@-@VERSION@
as they are revalidated.
.\"
.\"
.SH STATUS
.PP
The
.B @PACKAGE_TILE@
package is
.I production
grade on
.B Linux Fast-STREAMS
for the modules drivers included in the public release.  Not all OpenSS7
Project X.25 modules and drivers have been included in the release
distribution.  The OpenSS7 project is in the process or revalidating all
project modules and drivers for use on
.BR "Linux Fast-STREAMS" .
As a result, additional release components will be strapped into the package
and once full validation test suites are developed and run against each
component.
.PP
In particular, the following modules and drivers are currently missing format
the public release.
.\"
.\"
.[
$LIST$
.]
.TI


NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/util/init/strx25.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strx25:	start and stop strx25 subsystem
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strx25
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		character devices are present in the /dev directory and \
#		that the STREAMS X.25 subsystem is configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strx25
# Required-Start: streams $network
# Required-Stop: streams $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS X.25 subsystem
# License: GPL
# Description:	This STREAMS X.25 init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS X.25 character
#	devices are present in the /dev directory and that the STREAMS X.25
#	subsystem is configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strx25'
config="/etc/default/$name"
desc="the STREAMS X.25 subsystem"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRX25_MKNOD" ; then
    for STRX25_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $STRX25_MKNOD ] ; then
	    break
	else
	    STRX25_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$STRX25_PRELOAD"       ] || STRX25_PRELOAD=""
[ -n "$STRX25_DRIVERS"       ] || STRX25_DRIVERS="streams-x25-lapb streams-x25-plp"
[ -n "$STRX25_MODULES"       ] || STRX25_MODULES="streams-xot"
[ -n "$STRX25_MAKEDEVICES"   ] || STRX25_MAKEDEVICES="yes"
[ -n "$STRX25_REMOVEDEVICES" ] || STRX25_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRX25_MKNOD" ] && STRX25_MAKEDEVICES="no"
[ -z "$STRX25_MKNOD" ] && STRX25_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRX25_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRX25_MKNOD" -a ":$STRX25_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS X.25 devices: "
	$STRX25_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRX25_MKNOD" -a ":$STRX25_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS X.25 devices: "
	$STRX25_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRX25_PRELOAD $STRX25_DRIVERS $STRX25_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.2  2008-10-21 22:42:22  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.1  2008-05-03 10:46:39  brian
# - added package files
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
